lib/nose/statements.rb

```diff
@@ -49,8 +49,8 @@ module StatementConditions
     def populate_conditions(params)
       @conditions = params[:conditions]
       @eq_fields = conditions.each_value.reject(&:range?).map(&:field).to_set
-      @range_field = conditions.each_value.find(&:range?)
-      @range_field = @range_field.field unless @range_field.nil?
+      @range_fields = conditions.each_value.select(&:range?)
+      @range_fields = @range_fields.map{|f| f.field} unless @range_fields.nil?
```

Comment:
I don't think `@range_fields` will ever be `nil` since you're just filtering an array. These two lines could probably be simplified to

`@range_fields = conditions.each_value.select(&:range?).map(&:field)`